Docs/Design/Server_Tool_Executor_Phase2_Chat_AutoExec.md

```
@@ -0,0 +1,346 @@
+# result from the first execution — the note is created only once.
+
+# With chat_tool_idempotency = false
+# Auto-exec sends to MCP:
+#   tools/call { name: "notes.create",
+#                arguments: { title: "Meeting notes", body: "..." },
+#                idempotencyKey: null }
+#
+# Every invocation creates a new note, even if arguments are identical.
+```
+
+## Non-Streaming Flow (target behavior)
```

markdownlint-cli2 (0.20.0)

[warning] Headings should be surrounded by blank lines
Expected: 1; Actual: 0; Below

(MD022, blanks-around-headings)